Makefile

```diff
@@ -10,13 +10,33 @@ SHELL := /bin/bash
 DATA_PATH ?= $(shell pwd)/data
 # Set the environment variable MJKEY with the contents of the file specified by
 # MJKEY_PATH.
-MJKEY_PATH ?= ~/.mujoco/mjkey.txt
-
-test:  ## Run the CI test suite
-test: RUN_CMD = nice -n 11 pytest -v -n auto -m 'not huge and not flaky' --durations=0
+MJKEY_PATH ?= ${HOME}/.mujoco/mjkey.txt
+
+# BuildKit is required for the --secret flag in docker. We use it to mount mjkey.txt during build time
+export DOCKER_BUILDKIT=1
+
+build-test: TAG ?= rlworkgroup/garage-test-18.04
+build-test: docker/Dockerfile
+	docker build \
+		-f docker/Dockerfile \
+		--target garage-test-18.04 \
+		--secret id=mjkey,src=$(MJKEY_PATH) \
+		-t rlworkgroup/garage-test-18.04 \
+		${BUILD_ARGS} \
+		.
+
+test:  ## Run the garage-test-18.04 docker target
 test: RUN_ARGS = --memory 7500m --memory-swap 7500m
-test: run-headless
+test: CONTAINER_NAME ?= garage-test
```

Comment:
rather than overriding the container name, you should be able to capture it during build (return value) and re-use it during run.

Reply:
The container name is a runtime thing though. The image built during the build phase doesn't have a container name by itself.

Reply:
the current makefile sets a default container name, which you can override.
So you can do `make run-headless RUN_CMD="foo"` and it'll make a container called `garage-headless`
and then `CONTAINER_NAME='my-experiment-bar' make run-headless RUN_CMD="bar"` to make another container. And they'll run in parallel.

Reply:
OTOH, if we don't set any name by default, you can always do `make run-headless ...` and it'll create a new container with a random name, and they all run in parallel. But those names don't have any meaning. And your experiment directory would go in some directory like `$pwd/data/trusty_pangolin`.

Reply:
Yet another option is we keep the CONTAINER_NAME variable but set default value as empty instead of `garage-headless`, so you can override if you want, but you don't have to.

Reply:
perhaps we should remove naming the experiment directories by container name (since experiments and containers are usually 1:1 anyway)

i would advocate a default-empty container name, so that our system usually functions like docker normally would (people don't usually set container names)